documenting System.DllNotFoundException troubleshooting (#1085)
gircore/gir.core: docs/docs/faq.md

Common questions which can come up during development.

## DLL not found Exception

The `System.DllNotFoundException` can be thrown in the following cases:

- required DLLs or shared objects (the Unix equivalent of DLLs) are not installed in the operating system.

> - or `Module.Initialize()` was not called before instantiating an object in a namespace like `Gtk`, `Gdk` or `GtkSource`.

Reviewer: ```suggestion
- `Module.Initialize()` was not called before instantiating an object in a namespace like `Gtk`, `Gdk` or `GtkSource`.
```

> - required DLLs or shared objects are installed, but still `DllNotFoundException` is thrown. In this case, the names of the installed libraries probably don't match the names expected by `gir.core`. This can happen when using a custom build binary.

Reviewer: ```suggestion
- required DLLs or shared objects are installed, but still `DllNotFoundException` is thrown. In this case, the names of the installed libraries probably don't match the names expected by `gir.core`. This can happen when [using a custom build binary](#how-can-i-use-gircore-with-a-custom-build-native-binary).
```

> ## How can I use gir.core with a custom binary build of Gtk?

Reviewer: ```suggestion
## How can I use gir.core with a custom build native binary?
```

Common questions which can come up during development.

## DLL not found Exception

The `System.DllNotFoundException` can be thrown in the following cases:

> - required DLLs or Shared Objects (the Unix equivalent of DLLs) are not installed in the OS.

Reviewer: - Shared objects should probably be lower case?
- OS -> operating system

> - or you forgot to do a `Module.Initialize()` call before instantiating an object in a namespace like `Gtk`, `Gdk` or `GtkSource`.

Reviewer: - Please don't talk to the reader directly / avoid words like `you`
- Remove empty lines between bullet points

- required DLLs or Shared Objects are installed, but still `DllNotFoundException` is thrown. In this case, the names of the installed libraries probably don't match the names expected by `gir.core`. This can happen when using a custom build binary.

## How can I use gir.core with a custom binary build of Gtk?

The `gir.core` nuget packages are built against 3 different package sources:
1. Gnome SDK (Linux)
2. MSYS2 (Windows)
3. Homebrew (MacOS)

> Each of those sources defines the names of the binaries which must be available to call into them. If a custom build binary is used, the resulting binary name binary may be different from the one specified by the package source, resulting in a `System.DllNotFoundException`.

Reviewer: The sentence contains an obsolete `binary`.
